service/telemetry.go

```diff
@@ -57,6 +57,11 @@ const (
 	// supported trace propagators
 	traceContextPropagator = "tracecontext"
 	b3Propagator           = "b3"
+
+	// Default value of "service.name" resource attribute in the telemetry
+	// that the Collector produces about itself. This can be overridden
+	// by the end user via telemetry.Config.Resource setting.
+	defaultServiceName = "io.opentelemetry.collector"
```

Comment:
How can downstream distributions customize this?

Reply:
Distributions can have their own "confmap.Converter` that can change the config. See https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/converter.go

Reply:
While that's a powerful tool, I believe this will be such a common thing to override that we might want to provide an easier way to customize it?

Reply:
The build flag? that is very easy.

Also I don't think will be that common.

Reply:
> Also they can have a build ld flag to change this constant correct?

Constants cannot be overridden with build flags AFIAK, only variables (even internal)

I also think it might be useful to have a cleaner way to override this in distros

Reply:
We jump from 0 -> 1000 here. We did not have any car before now we want a 2M$ car. Let's see if we have this request before any more public API. We provided a way via config. Also another way is to do via config flags ("--set"), there are at least 3 different ways to achieve this, and we are asking for more.

Also we are talking about distros (2-3), for that if this is not the cleanest solution for the moment is ok, final users should have a clean solution and they have.